users/user-models.js

```diff
@@ -1,35 +1,39 @@
 var mongoose = require('mongoose');
 var Schema = mongoose.Schema;
 
+var options = {discriminatorKey: 'type'};
```

Comment:
I would make the ```discriminatorKey``` the ```role``` field, since that's the field that would normally show whether the user is a student or admin.  Also, having both a ```role``` and ```type``` field would be duplicating the role information.